controllers/admin/AdminSmartsuppAjax.php

```diff
@@ -32,26 +32,36 @@ public function init()
                 
         switch (Tools::getValue('action')) {
             case 'login':
-                $response = $api->login(array(
+                $response = $api->login([
                     'email' => Tools::getValue('email'),
                     'password' => Tools::getValue('password'),
                     'platform' => 'Prestashop ' . _PS_VERSION_,
-                ));
+                ]);
 
-                Configuration::updateValue('SMARTSUPP_KEY', $response['account']['key']);
-                Configuration::updateValue('SMARTSUPP_EMAIL', Tools::getValue('email'));
+                if (!isset($response['account']['key'])) {
+                    $response['error'] = isset($response['error']) ? $response['error'] : $this->module->l('Unknown Error Occurred');
+                    $response['message'] = isset($response['message']) ? $response['message'] : $this->module->l('Unknown Error Occurred');
```

Comment:
self::FILE_NAME for translatable text

Reply:
to skip usage of else, might as well break after filling in the response. Would make this a early return